src/Security/Authentication/Token/LegacyOAuth2Token.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace League\Bundle\OAuth2ServerBundle\Security\Authentication\Token;
+
+use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Component\Security\Core\Authentication\Token\AbstractToken;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+final class LegacyOAuth2Token extends AbstractToken
+{
+    /**
+     * @var string
+     */
+    private $providerKey;
+
+    public function __construct(
+        ServerRequestInterface $serverRequest,
+        ?UserInterface $user,
+        string $rolePrefix,
+        string $providerKey
+    ) {
+        $this->setAttribute('server_request', $serverRequest);
```

Comment:
this is not a solution to #17, attributes will be serialized, if the request contains a file, it won't be serializable.

Reply:
Yes indeed.
It's only when using the authenticator security system that #17 is solved.
We still have to find a way to fix it when using the legacy security system

Reply:
let's drop it then? we shouldn't care about BC :)

Reply:
Actually, I don't know if we wanna keep the support for the legacy security system which is (and will be) mainly used.
Any thoughts @chalasr?

Reply:
Let's drop it :) Sorry if I was unclear. 
(FYI the new system is considered stable as of Symfony 5.3, and the old systems (guard included) are going to be deprecated.)